frontend/src/components/graduations/GraduationCard.tsx

```diff
@@ -0,0 +1,373 @@
+'use client';
+
+import { useAuth } from '@/auth/Auth';
+import { Graduation } from '@/database/graduation';
+import { formatRatingSystem, RatingSystem } from '@/database/user';
+import CohortIcon from '@/scoreboard/CohortIcon';
+import { ChessDojoIcon } from '@/style/ChessDojoIcon';
+import { RatingSystemIcon } from '@/style/RatingSystemIcons';
+import { SaveAlt } from '@mui/icons-material';
+import ArrowDownwardIcon from '@mui/icons-material/ArrowDownward';
+import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward';
+import { LoadingButton } from '@mui/lab';
+import { Box, Stack, Typography, useTheme } from '@mui/material';
+import { domToPng } from 'modern-screenshot';
+import {
+    ForwardedRef,
+    forwardRef,
+    ReactNode,
+    useEffect,
+    useMemo,
+    useRef,
+    useState,
+} from 'react';
+import { AxisOptions, Chart, UserSerie } from 'react-charts';
+interface Datum {
+    date: Date;
+    rating: number;
+}
+
+export const primaryAxis: AxisOptions<Datum> = {
+    scaleType: 'time',
+    getValue: (datum) => datum.date,
+};
+
+export const secondaryAxes: AxisOptions<Datum>[] = [
+    {
+        scaleType: 'linear',
+        getValue: (datum) => datum.rating,
+        formatters: {
+            scale: (value) => `${value}`,
+        },
+    },
+];
+
+interface ReportCanvasProps {
+    width?: number | string;
+    height?: number | string;
+    children: React.ReactElement;
+}
+
+const ReportCanvas = forwardRef(function ReportCanvas(
+    { children }: ReportCanvasProps,
+    ref: ForwardedRef<HTMLDivElement>,
+) {
+    return (
+        <Box
+            ref={ref}
+            display='grid'
+            sx={{
+                width: '800px',
+                height: '500px',
+            }}
+        >
+            {children}
+        </Box>
+    );
+});
+
+interface GraduationCardProps {
+    graduation: Graduation;
+}
+
+function getChartData(graduation: Graduation): UserSerie<Datum>[] {
+    const { ratingSystem: preferredSystem, startedAt } = graduation;
+
+    const ratingHistory =
+        graduation.ratingHistories?.[preferredSystem]
+            ?.filter(
```

Comment:
Is this filter necessary? Aren't you already doing this on the backend?

Reply:
In order to support rapid cohort advancement, I was going to grab the user's whole history if the graduation object for that cohort has none.